portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,37 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+
+/** Load a random pic for a specific splash page icon
+ */
+function randomPic(filePrefix, numFiles, elId) {
+  console.log("For " + filePrefix + " elId is" + elId);
+  var picOptions = makeFileArray(filePrefix, numFiles); //create arr of options
+  console.log("picOptions are " + picOptions);
+  randInd = Math.floor((Math.random() * picOptions.length));
+
+  document.getElementById(elId).src = picOptions[randInd];
+  console.log("for " + elId + " picked image " + picOptions[randInd]);
+  console.log("\n");
+  return true;
+}
+
+/** makes an array where arr[i] = filePrefix_i.jpeg
+ */
+function makeFileArray(filePrefix,numFiles) {
+  var arr = new Array(numFiles);
+  for(var i = 0; i < numFiles; i++) {
+    arr[i] = filePrefix + "_" + (i+1) + ".jpeg"; //files are 1-indexd
+  }
+  return arr;
+}
+/** Map a relevant randomPic to each splash page tiles
+ */
+function randomPics() {
+  const tiles = [["hiking",8],["me",9],["projects",12],["travel",8]];
+  const _     = tiles.map(x => randomPic("images/" + x[0] + "_images/"+x[0], x[1], x[0] + "RandPic"));
```

Comment:
The first parameter here is complicated and makes the line hard to read.  For clarity, declare it as a separate variable.

Also this would be much easier to parse if x was a json object and had a more descriptive name than x :)

Reply:
agreed, rewritten for clarity